doc/smart-switch/BFD/SmartSwitchDpuLivenessUsingBfd.md

```diff
@@ -0,0 +1,265 @@
+
+Trigger for HA manager to create BFD sessions will be detailed in HA HLD.
+
+<img src="images/Dash_BFD_FRR.png">
+
+BFD passive sessions in DPU waits for BFD control packets from NPU's
+active session. When FRR stack receives BFD packets from active sessions
+in NPU, it starts responding. After the initial handshake, BFD sessions
+are up. When active session doesn't receive control packets for the
+configured BFD timeout period, session is set to DOWN and corresponding
+action is taken.
+
+<img src="images/NPU_DPU_BFD_Exchange.png">
+
+## 2.4 DPU FRR Changes
+
+By default BFD Daemon is not enabled in FRR. FRR supervisor config file
+has to be updated to start BFDD during FRR initialization.
+
+To configure BFD passive sessions, a profile with 'passive' mode is
+created and used for all sessions created in FRR. BgpCfgd configures FRR
+stack using vtysh.
+
+```
+
+bfd
+
+profile passive
+
+passive-mode
+
+receive-interval 100
+
+transmit-interval 100
+
+detect multiplier 3
+
+exit
+
+!
+
+peer \<Local/Remote NPU IP\> local-address \<DPU PA\>
+
+profile passive
+
+exit
+```
+
+## 2.5 DPU Linux IPTables
+
+New rules need to be added in the kernel to allow BFD packets to be
+processed by the kernel and FRR stack. All packets matching BFD UDP(3784
+and 4784).
+
+***sudo iptables \--insert INPUT 5 -p udp \--dport 4784 -j ACCEPT***
+
+***sudo iptables \--insert INPUT 5 -p udp \--dport 3784 -j ACCEPT***
+
+These rules can be added as part of caclmgrd script during initialization by checking the DEVICE_METADATA object and attribute "switch_type: DPU" in CONFIG_DB.
+
+## 2.6 NPU BFD session and VNET_ROUTE_TUNNEL
+
+Overlay ECMP HLD describes how BFD session UP/Down changes the route to
+VIP of a SDN appliance. In Overlay ECMP feature, all the nexthops are
+VxLAN tunnel nexthops and the route to VIP is updated with primary or
+secondary ECMP group based on the BFD session up/down state change. In
+Smartswitch deployment, when a DPU VIP has nexthop local to
+NPU, nexthop will be a regular IP nexthop instead of a VxLAN NH.
+
+When HA manager receives HA config update, it will create VNET Route tunnel table and that triggers BFD session creation in NPU. These BFD sessions are hardware offloaded to NPU.
+
+<img src="images/DB_And_Notification.png">
+
+## 2.7 NPU-DPU midplane state change and BFD session update
+
+When NPU-DPU midplane interface goes down due to some trigger or platform changes, PMON module updates the midplane NetDev interface state in DPU_STATE DB and trigger a notification for link state change. HA manager listens to these updates and bring down the BFD by applying ingress ACL rules or updating NPU->DPU links. This will bring down the local and remote BFD sessions between NPU and DPU. ACL can be attached to NPU-DPU interconnect interface to drop BFD packets from DPU.
+
+Note: This is out of scope of this document, Smartswitch HA HLD will have more details as HA manager listens to all FSM changes from the controller and PMON changes.
+

```

Comment:
we can refer the Unplanned Events section in HA HLD for more details, if needed.